src/transformStateWithClones.js

```diff
@@ -6,8 +6,79 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+
+  let currentState = { ...state };
+
+  actions.forEach((action) => {
+    currentState = { ...currentState };
```

Comment:
[CODE STYLE] - Don't repeat yourself by copying the currentState inside the forEach loop for every action. Instead, do it just once after the switch statement, before pushing the currentState to the stateHistory array.